frontend/apps/reader/modules/readerannotation.lua

```diff
@@ -254,6 +254,12 @@ function ReaderAnnotation:updatePageNumbers(force_update)
     self.needs_update = nil
 end
 
+function ReaderAnnotation:onUsePageLabelsUpdated()
+    self:updatePageNumbers(true)
+end
+
+ReaderAnnotation.onUsePageLabelsUpdated = ReaderAnnotation.setNeedsUpdateFlag
```

Comment:
Please put it right after
https://github.com/koreader/koreader/blob/1bdf0dd0a4a331a6d6d393ab4900b4a0ca4aba29/frontend/apps/reader/modules/readerannotation.lua#L242